mava/systems/q_learning/sebulba/rec_iql.py

```diff
@@ -0,0 +1,782 @@
+    if config.logger.checkpointing.load_model:
+        loaded_checkpoint = Checkpointer(
+            model_name=config.logger.system_name,
+            **config.logger.checkpointing.load_args,  # Other checkpoint args
+        )
+        # Restore the learner state from the checkpoint
+        restored_params, _ = loaded_checkpoint.restore_params(input_params=params)
+        # Update the params
+        params = restored_params
+
+    # Duplicate learner across Learner devices.
+    params, opt_state = jax.device_put((params, opt_state), learner_sharding)
+
+    # Initial learner state.
+    init_learner_state = LearnerState(params, opt_state, 0)
+
+    env.close()
+    return learn, q_net, init_learner_state, learner_sharding, init_transition
+
+
+def run_experiment(_config: DictConfig) -> float:
+    """Runs experiment."""
+    config = copy.deepcopy(_config)
+
+    local_devices = jax.local_devices()
+    devices = jax.devices()
+    err = "Local and global devices must be the same, we dont support multihost yet"
+    assert len(local_devices) == len(devices), err
+    learner_devices = [devices[d_id] for d_id in config.arch.learner_device_ids]
+    actor_devices = [local_devices[device_id] for device_id in config.arch.actor_device_ids]
+
+    # JAX and numpy RNGs
+    key = jax.random.PRNGKey(config.system.seed)
+    np_rng = np.random.default_rng(config.system.seed)
+
+    # Setup learner.
+    learn, q_net, learner_state, learner_sharding, init_transition = learner_setup(
+        key, config, learner_devices
+    )
+
+    # Setup evaluator.
+    # One key per device for evaluation.
+    eval_act_fn = make_rec_eval_act_fn(q_net.apply, config)
+
+    evaluator, evaluator_envs = get_eval_fn(
+        environments.make_gym_env, eval_act_fn, config, np_rng, absolute_metric=False
+    )
+
+    # Calculate total timesteps.
+    config = check_total_timesteps(config)
+    check_sebulba_config(config)
+
+    steps_per_rollout = (
+        config.system.rollout_length
+        * config.arch.num_envs
+        * config.system.num_updates_per_eval
+        * len(config.arch.actor_device_ids)
+        * config.arch.n_threads_per_executor
+    )
+
+    # Setup RateLimiter
+    insert_to_sample_ratio = (
+        config.system.rollout_length
+        * config.arch.num_envs
+        * len(config.arch.actor_device_ids)
+        * config.arch.n_threads_per_executor
+    ) / (config.system.sample_sequence_length * config.system.sample_batch_size)
+
+    config.sample_per_insert = config.system.data_sample_mean * insert_to_sample_ratio
+    config.tolerance = config.sample_per_insert * config.system.error_tolerance
+
+    min_num_inserts = max(
+        config.system.sample_sequence_length // config.system.rollout_length,
+        config.system.min_buffer_size // config.system.rollout_length,
+        1,
+    )
+
+    rate_limiter = SampleToInsertRatio(config.sample_per_insert, min_num_inserts, config.tolerance)
```

Comment:
We should probably put them in the system config so it's easier to find on things like neptune

```suggestion
    config.system.sample_per_insert = config.system.data_sample_mean * insert_to_sample_ratio
    config.system.tolerance = config.sample_per_insert * config.system.error_tolerance

    min_num_inserts = max(
        config.system.sample_sequence_length // config.system.rollout_length,
        config.system.min_buffer_size // config.system.rollout_length,
        1,
    )

    rate_limiter = SampleToInsertRatio(config.system.sample_per_insert, min_num_inserts, config.system.tolerance)
```